src/fides/api/ops/service/connectors/consent_email_connector.py

```diff
@@ -0,0 +1,160 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+from sqlalchemy.orm import Query, Session
+
+from fides.api.ctl.sql_models import Organization  # type: ignore[attr-defined]
+from fides.api.ops.common_exceptions import MessageDispatchException
+from fides.api.ops.models.connectionconfig import (
+    AccessLevel,
+    ConnectionConfig,
+    ConnectionTestStatus,
+    ConnectionType,
+)
+from fides.api.ops.schemas.connection_configuration import ConsentEmailSchema
+from fides.api.ops.schemas.connection_configuration.connection_secrets_email_consent import (
+    SVORN_REQUIRED_IDENTITY,
+    AdvancedSettings,
+)
+from fides.api.ops.schemas.messaging.messaging import (
+    ConsentEmailFulfillmentBodyParams,
+    ConsentPreferencesByUser,
+    MessagingActionType,
+)
+from fides.api.ops.schemas.privacy_request import Consent
+from fides.api.ops.schemas.redis_cache import Identity
+from fides.api.ops.service.connectors.base_connector import LimitedConnector
+from fides.api.ops.service.messaging.message_dispatch_service import dispatch_message
+from fides.core.config import get_config
+
+CONFIG = get_config()
+
+CONSENT_EMAIL_CONNECTOR_TYPES = [ConnectionType.sovrn]
+
+
+class GenericEmailConsentConnector(LimitedConnector[None]):
+    """Generic Email Consent Connector that can be overridden for specific vendors"""
+
```

Comment:
This ticket does not expose a "generic" consent email connector but I put in the foundation to do this later.   I've created a generic connector here, that the Sovrn connector extends.  When we later want to surface a generic consent email connector type that works with any service, we could just have that connection config use this connector.